presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -626,8 +625,7 @@ private void invalidateTableStatisticsCache(String databaseName, String tableNam
     {
         return delegate.getPartition(
                 partitionName.getIdentity(),
-                partitionName.getKey().getHiveTableName().getDatabaseName(),
-                partitionName.getKey().getHiveTableName().getTableName(),
+                partitionName.getKey().getTable().get(),
```

Comment:
What if `! partitionName.getKey().getTable().isPresent()`?
What if it is stale?

Reply:
> What if ! partitionName.getKey().getTable().isPresent()?

It can't be empty as partitions cache always set `table` field.

> What if it is stale?

I think the assumption is that within transaction metadata doesn't really change. Otherwise exception should be thrown. Anyway, in current code metadata might be staled too (between `CachingHiveMetastore` and `ThriftMetastoreClient`)

Reply:
> It can't be empty as partitions cache always set `table` field.

then why do we need Optional?


> I think the assumption is that within transaction metadata doesn't really change.

note: this is not only about transactions, when caching enabled

Reply:
> then why do we need Optional?

`HivePartitionName` is used by different caches (e.g. stats) which provide different arguments or params. I expect this to simplify once we remove more metastore calls (e.g. for stats).

> note: this is not only about transactions, when caching enabled

In current code there might still be inconsistency (e.g. `getPartition` detects Avro format with table schema while `getTable` doesn't see that). Currently calls were leaking through session cache. This need to be fixed. If we find issues with long-term cache, we should consider changing strategy there.